tests/python/kivaloo/servers.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+""" Launch and organize servers for the python tests. """
+
+import logging
+import os
+import shutil
+import subprocess
+import threading
+import queue
+
+import psutil
+
+
+# ### Private constants
+_KIVALOO_TEST_DIR = "/tmp/kivaloo-test/"
+# - this is relative to TEST_DIR
+# - this may be subjected to rm -rf
+_DISK_LBS = "lbs-disk"
+
+# These filenames are relative to this directory
+_BIN_LBS = "../../lbs/lbs"
+_BIN_KVLDS = "../../kvlds/kvlds"
+
+# ### Public constants
+LBS_BLOCKSIZE = 512
+
+LBS_SOCK = os.path.join(_KIVALOO_TEST_DIR, "kivaloo-lbs-sock")
+KVLDS_SOCK = os.path.join(_KIVALOO_TEST_DIR, "kivaloo-kvlds-sock")
+
+
+def _get_process(cmdline_split):
+    """ Find the PID corresponding to a command-line. """
+    for pid in psutil.process_iter():
```

Comment:
Uhh...why???  The kivaloo damons write their pids to files...

Reply:
1. this should actually be `proc`, not `pid`.  That's a minor documentation problem which I'll fix.

2. sometimes naughty people kill the lbs server without deleting the pidfile.  Can we assume that if a pidfile exists, there is a server at the corresponding pid?  More importantly, can we assume that there isn't a different process that's reused that pid?

    If there's a process with exactly the same command-line as the test, I'm comfortable killing that process.  Otherwise, it seems like it's asking for trouble.

Reply:
Also, it's not just a question of being "naughty".  When I'm poking unfamiliar code with a stick, I tend to be rather free with `printf("blah\n"); exit(1);`.  So the test suite should be robust against Graham-induced failures in the servers.

Reply:
It definitely makes sense to check that the process is what we think is is before we kill it.  But we should start with the pidfile, not just scan the entire system.